pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -511,96 +521,80 @@ contract StablePhantomPool is StablePool, ProtocolFeeCache {
     {
-        StablePhantomPoolUserData.ExitKindPhantom kind = userData.exitKind();
-
-        // Exits typically revert, except for the proportional exit when the emergency pause mechanism has been
-        // triggered. This allows for a simple and safe way to exit the Pool.
-        if (kind == StablePhantomPoolUserData.ExitKindPhantom.EXACT_BPT_IN_FOR_TOKENS_OUT) {
-            _ensurePaused();
-
-            // Note that this will cause the user's BPT to be burned, which is not something that happens during
-            // regular operation of this Pool, and may lead to accounting errors. Because of this, it is highly
-            // advisable to stop using a Pool after it is paused and the pause window expires.
-
-            (bptAmountIn, amountsOut) = _proportionalExit(balances, userData);
-            // For simplicity, due protocol fees are set to zero.
-            dueProtocolFeeAmounts = new uint256[](_getTotalTokens());
-        } else {
-            _revert(Errors.UNHANDLED_BY_PHANTOM_POOL);
-        }
+        _revert(Errors.UNHANDLED_BY_PHANTOM_POOL);
     }
 
-    function _proportionalExit(uint256[] memory balances, bytes memory userData)
-        private
-        view
-        returns (uint256, uint256[] memory)
-    {
-        // This proportional exit function is only enabled if the contract is paused, to provide users a way to
-        // retrieve their tokens in case of an emergency.
-        //
-        // This particular exit function is the only one available because it is the simplest, and therefore least
-        // likely to be incorrect, or revert and lock funds.
+    // We cannot use the default implementation here, since we need to account for the BPT token
+    function _doRecoveryModeExit(
+        uint256[] memory balances,
+        uint256,
+        bytes memory userData
+    ) internal virtual override returns (uint256, uint256[] memory) {
         (, uint256[] memory balancesWithoutBpt) = _dropBptItem(balances);
 
-        uint256 bptAmountIn = userData.exactBptInForTokensOut();
+        uint256 bptAmountIn = userData.recoveryModeExit();
         // Note that there is no minimum amountOut parameter: this is handled by `IVault.exitPool`.
 
         uint256[] memory amountsOut = StableMath._calcTokensOutGivenExactBptIn(
             balancesWithoutBpt,
             bptAmountIn,
-            // This process burns BPT, rendering the approximation returned by `_dropBPTItem` inaccurate,
-            // so we use the real method here
             _getVirtualSupply(balances[_bptIndex])
         );
 
         return (bptAmountIn, _addBptItem(amountsOut, 0));
     }
 
+    // Override the implementation in StablePool, which has special processing to "reset" protocol fee
+    // calculation after disabling recovery mode. This is unnecessary here, as protocol fees are
+    // acccumulated on each swap (or not, if recovery mode is enabled).
+    function _setRecoveryMode(bool enabled) internal virtual override {
```

Comment:
Good catch. We'll be able to get rid of these things thankfully once we remove StablePool. If we chose to kept it, we could also introduce a BaseStablePool, so this doesn't seem too concerning.